test-data/unit/check-final.test

```diff
@@ -339,22 +339,29 @@ class C:
 from typing import Final
 
 x: Final = 1
+x
```

Comment:
I strongly prefer to not shadow final names, since this kind of defeats the purpose.

Reply:
My implementation does some additional checks in a later pass after renaming to prevent the redefinition of final names. It should preserve the existing semantics of final names. Or is there something else you are worried about?

Reply:
The `x` is reference is used to trigger the renaming machinery. Otherwise the second definition would be trivially rejected, as it would not count as a new definition. My implementation only renames a variable if it has been read, since this shouldn't be treated as a redefinition:

```py
x = None  # type: List[int]
x = []  # Not a redefinition
```